### PR TITLE
[feat/fix]: Improve the contributing guidelines

### PR DESCRIPTION
**Pull request by Panquesito7**
#### Description of Change

- Improve the contributing guidelines.
  - Mostly fixes grammar, spelling errors, typos, and improves wording.

#### Checklist

- [x] Added description of change
- [ ] ~Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)~
- [ ] ~Added tests and example, test must pass~
- [ ] ~Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)~
- [ ] ~Relevant documentation/comments is changed or added~
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1503"></a>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
 
 ### Contributor
 
-We are very happy that you consider implementing algorithms and data structures for others! This repository is referred to and used by learners from around the globe. Being one of our contributors, you agree and confirm that:
+We are happy that you consider implementing algorithms and data structures for others! This repository is referred to and used by learners from around the globe. Being one of our contributors, you agree and confirm that:
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:11]:
```suggestion
Being a contributor at The Algorithms we request you to follow the points mentioned below.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
 
 - You did your own work.
   - No plagiarism allowed. Any plagiarized work will not be merged.
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:14]: - Please try to provide your own implementations with proper documentation. We do not entertain any plagiarized work.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
 - Your work will be distributed under [MIT License](License) once your pull request has been merged.
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:15]: This looks good.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
-- You submitted work fulfils or mostly fulfils our styles and standards.
+- Your submitted work fulfills or mostly fulfills our styles and standards.
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:16]:
```suggestion
- Please follow the repository coding guidelines (mentioned below)
```

Comment (ayaankhan98) [reply CONTRIBUTING.md:16]: Also add one more point
- Please make sure that you do not provide any duplicate work (in a sense that implementation of data structure or algorithm is already present in repository)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
 
-**New implementation** New implementation are welcome!
+**New implementation** New implementations are welcome!
 
 **Improving comments** and **adding tests** to existing algorithms are much appreciated.
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:20]: - You can add new algorithms or data structures which are not present in repository or can improve the old implementations (documentation, improving test cases, removing bugs or in any other resonable sense)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
 
-**Issues** Please avoid opening issues asking to be "assigned” to a particular algorithm.  This merely creates unnecessary noise for maintainers.  Instead, please submit your implementation in a pull request and it will be evaluated by project maintainers.
+**Issues** Please avoid opening issues asking to be "assigned” to a particular algorithm. This merely creates unnecessary noise for maintainers. Instead, please submit your implementation in a pull request, and it will be evaluated by project maintainers.
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:22]: This looks good.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
 
 ### Making Changes
 
 #### Code
 
 - Please use the directory structure of the repository.
-- File extension for code should be `*.hpp`, `*.h` or `*.cpp`.
-- Don't use **bits/stdc++.h** because this is quite Linux specific and slows down the compilation process.
-- Organize your code using **`struct`**, **`class`** and/or **`namespace`** keywords
+- File extension for the code should be `*.hpp`, `*.h` or `*.cpp`.
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:29]:
```suggestion
- Make sure file extensions should be `*.hpp`, `*.h` or `*.cpp`.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
 - If an implementation of the algorithm already exists, please refer to the [file-name section below](#new-file-name-guidelines).
 - You can suggest reasonable changes to existing algorithms.
 - Strictly use snake_case (underscore_separated) in filenames.
 - If you have added or modified code, please make sure the code compiles before submitting.
-- Our automated testing runs [__CMake__](https://cmake.org/) on all pull requests so please be sure that your code passes before submitting.
-- Please conform to [doxygen](https://www.doxygen.nl/manual/docblocks.html) standard and document the code as much as possible. This not only facilitates the readers but also generates the correct info on website.
-- **Be consistent in use of these guidelines.**
+- Our automated testing runs [__CMake__](https://cmake.org/) on all the pull requests, so please be sure that your code passes before submitting.
+- Please conform to [Doxygen](https://www.doxygen.nl/manual/docblocks.html) standard and document the code as much as possible. This not only facilitates the readers but also generates the correct info on the website.
+- **Be consistent in the use of these guidelines.**
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:38]: Looks good

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -8,48 +8,48 @@ Welcome to [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-P
...
 
 #### Documentation
 
-- Make sure you put useful comments in your code.  Do not comment things that are obvious.
+- Make sure you put useful comments in your code. Do not comment on obvious things.
 - Please avoid creating new directories if at all possible. Try to fit your work into the existing directory structure. If you want to create a new directory, then please check if a similar category has been recently suggested or created by other pull requests.
 - If you have modified/added documentation, please ensure that your language is concise and contains no grammar errors.
```
Comment (ayaankhan98) [inline CONTRIBUTING.md:44]:
```suggestion
- If you have modified/added documentation, please ensure that your language is concise and must not contain gramatical errors.
```

**Review by ayaankhan98 — COMMENTED** [review]
just given few suggestions, these are not mandatory but are suggestions if you like then we can change on these things too.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Panquesito7) [reply CONTRIBUTING.md:14]: I don't think this is needed. The current paragraph is OK IMO.

File: CONTRIBUTING.md (hunk 4 above)
Comment (Panquesito7) [reply CONTRIBUTING.md:16]: > Suggestion

What about `Please follow the repository guidelines and standards mentioned below`?

**Review by ayaankhan98 — APPROVED** [review]

**Pull request merged**